### PR TITLE
adding readme file to examples/api

### PR DESCRIPTION
**Pull request by tohar-ayash**
Signed-off-by: tohar-ayash <toharayash1@gmail.com>

### CODE REVIEW

File: examples/api/README.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+
```
Comment (tock-ibm) [inline examples/api/README.md:1]: Start with a description of what's in here... for example:
`The examples folder contain... bla bla... here we describe how to.... etc`

File: examples/api/README.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+## Build and run examples inside Docker
+### Prerequisites
+
+To build a docker image, the following are the prerequisites that should be installed.
+
+- **Docker**: To generate cryptographic materials and build BCDB image.
```
Comment (tock-ibm) [inline examples/api/README.md:7]: not accurate, docker is used to run the BCDB image and will also contain the crypto material.
change the wording....

File: examples/api/README.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+- **Git**: To clone the code repository.
+
+### Build
+
+Build process includes two steps - crypto materials generation and docker image build
```
Comment (tock-ibm) [inline examples/api/README.md:12]: ->The build process

provide some notes about what is the right version of the server that runs with the sdk, we will have to formalize this some how... for now its better to just remark that the main head is a development branch, not always compatible with the sdk.

File: examples/api/README.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+```
+cd bcdb-server
```
Comment (tock-ibm) [inline examples/api/README.md:15]: should we not clone the server first to an adjacent dir next to bcdb-sdk? 
detail the assumption about the dir structure you expect.

File: examples/api/README.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+```
+
+#### Generate cryptographic materials
+   
+Generate crypto materials for users used in the examples
+```
+./scripts/cryptoGen.sh sampleconfig alice bob
```
Comment (tock-ibm) [inline examples/api/README.md:22]: why "sampleconfig" and not "server"?

File: examples/api/README.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+Generate crypto materials for users used in the examples
+```
+./scripts/cryptoGen.sh sampleconfig alice bob 
+```
+
+#### Generate docker image
+```
+make docker
+```
+#### Start docker container
```
Comment (tock-ibm) [inline examples/api/README.md:29]: Start the server inside a docker container

File: examples/api/README.md [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+```
+docker run -it --rm -v $(pwd)/sampleconfig/:/etc/bcdb-server -p 6001:6001 -p 7050:7050 bcdb-server
+```
```
Comment (tock-ibm) [inline examples/api/README.md:32]: say something on whether the build steps should be repeated for each example? how do we reset the server after running one example and preparing to run another one?

File: examples/api/README.md [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+``` 
+Build and run  
+``` 
+go build
+``` 
+``` 
+./<example-dir>
+``` 
+
+
```
Comment (tock-ibm) [inline examples/api/README.md:49]: Add a section about clean up after running examples, which data directories need to be deleted?

And test all the steps above, running multiple examples one after the other...

File: examples/api/README.md (hunk 3 above)
Comment (tock-ibm) [reply examples/api/README.md:12]: This appears the `go.mod` file, e.g.
```
require (
	github.com/IBM-Blockchain/bcdb-server v0.1.1-0.20210815133509-935d047b0059
...
)
```

File: examples/api/README.md (hunk 5 above)
Comment (gennadylaventman) [reply examples/api/README.md:22]: First argument is folder.

**Review by tock-ibm — APPROVED** [review]

**Pull request merged**